src/js/utils/ui.js

```diff
@@ -0,0 +1,142 @@
+define([
+    'utils/backbone.events',
+    'events/events',
+    'utils/underscore'
+], function(Events, events, _) {
+    var TOUCH_MOVE = 'touchmove',
+        TOUCH_START = 'touchstart',
+        TOUCH_END = 'touchend',
+        TOUCH_CANCEL = 'touchcancel',
+        MOUSE_DOWN = 'mousedown',
+        MOUSE_MOVE = 'mousemove',
+        MOUSE_UP = 'mouseup';
+
+    var ui = function (elem) {
+        var _elem = elem,
+            _isPressed = false,
+            _startEvent = null,
+            _hasMoved = false,
+            _lastClickTime = 0;
+
+        _.extend(this, Events);
+
+        elem.addEventListener(TOUCH_START, interactStartHandler);
+        elem.addEventListener(MOUSE_DOWN, interactStartHandler);
+
+        function interactStartHandler() {
+            _isPressed = true;
+
+            if(self._events[events.touchEvents.DRAG]){
```

Comment:
This seems functional, but really we want to pretend we don't know the _events object exists. It is an implementation detail of the backbone events object and is subject to change.